extension/goplugin/auth.go

```diff
@@ -55,32 +57,75 @@ func (a *Auth) IsAdmin(context goext.Context) bool {
 	return auth.IsAdmin()
 }
 
-func getRoleFromContext(context goext.Context) (*schema.Role, error) {
-	roleRaw, ok := context["role"]
+func (a *Auth) IsTenantIDValid(ctx goext.Context, id string) error {
+	identityService, err := getIdentityServiceFromContext(ctx)
+	if err != nil {
+		return err
+	}
+	return identityService.IsTenantIDValid(id)
+}
+
+func getIdentityServiceFromContext(ctx goext.Context) (middleware.IdentityService, error) {
+	rawIdentityService, err := getFromContext(ctx, "identity_service")
+	if err != nil {
+		return nil, err
+	}
+
+	identityService, ok := rawIdentityService.(middleware.IdentityService)
+	if !ok {
+		err := newInvalidTypeErr("identity_service")
+		log.Warning(err.Error())
```

Comment:
maybe add some context to this error

Reply:
MIO error itself carries enough context. I've refactored a little bit to remove duplication